src/chef_authn.erl

```diff
@@ -481,26 +534,26 @@ verify_sig_or_sigs(Plain, BodyHash, ContentHash, AuthSig, UserId, KeyData, SignV
 %% what we are repeating each step
 %%
 -spec verify_sigs(binary(), binary(), binary(), binary(), binary(),
-                  [{key_desc(), public_key_data() | public_key:rsa_public_key()}], binary()) ->
+                  [{key_desc(), public_key_data() | public_key:rsa_public_key()}], {binary(), binary()}) ->
                          {name, user_id(), key_desc()}.
 
-verify_sigs(Plain, BodyHash, ContentHash, AuthSig, UserId, [{KeyId, PubKey}], SignVersion) ->
-    {name, UserId} = verify_sig(Plain, BodyHash, ContentHash, AuthSig, UserId, PubKey, SignVersion),
+verify_sigs(Plain, BodyHash, ContentHash, AuthSig, UserId, [{KeyId, PubKey}], SignInfo) ->
+    {name, UserId} = verify_sig(Plain, BodyHash, ContentHash, AuthSig, UserId, PubKey, SignInfo),
     {name, UserId, KeyId};
-verify_sigs(Plain, BodyHash, ContentHash, AuthSig, UserId, [{KeyId, PubKey} | Remaining], SignVersion) ->
+verify_sigs(Plain, BodyHash, ContentHash, AuthSig, UserId, [{KeyId, PubKey} | Remaining], SignInfo) ->
     try
-        {name, UserId} = verify_sig(Plain, BodyHash, ContentHash, AuthSig, UserId, PubKey, SignVersion),
+        {name, UserId} = verify_sig(Plain, BodyHash, ContentHash, AuthSig, UserId, PubKey, SignInfo),
         {name, UserId, KeyId}
     catch
         error:_Any ->
-            verify_sigs(Plain, BodyHash, ContentHash, AuthSig, UserId, Remaining, SignVersion)
+            verify_sigs(Plain, BodyHash, ContentHash, AuthSig, UserId, Remaining, SignInfo)
     end.
 
 -spec verify_sig(binary(), binary(), binary(), binary(), binary(),
-                 public_key_data() | public_key:rsa_public_key(), binary()) ->
+                 public_key_data() | public_key:rsa_public_key(), {binary(), binary()}) ->
                         {name, user_id()}.
 
-verify_sig(Plain, BodyHash, ContentHash, AuthSig, UserId, PublicKey, SignVersion)
+verify_sig(Plain, BodyHash, ContentHash, AuthSig, UserId, PublicKey, {_, SignVersion})
   when SignVersion =:= ?SIGNING_VERSION_V1_0;
        SignVersion =:= ?SIGNING_VERSION_V1_1 ->
     Plain = decrypt_sig(AuthSig, PublicKey),
```

Comment:
This is old code I know, but any security equality check should be constant time; e.g. the sig and hash should take the same amount of time no matter how many bits differ.